### PR TITLE
sm64ex: All Bowser Stages Goal

### PR DESCRIPTION
**Pull request by N00byKing**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
New goal for sm64ex, instead of just BitS this requires completing both BitDW and BitFS as well.
Was asked for on discord, felt like adding it.

## How was this tested?
Generated, playtested.
If someone is interested in testing this, note that sm64ex needs to be compiled on branch `ap_test`

### CODE REVIEW

File: worlds/sm64ex/Rules.py [hunk 1]
```diff
@@ -122,4 +122,9 @@ def set_rules(world, player: int, area_connections):
     add_rule(world.get_location("MIPS 1", player), lambda state: state.can_reach("Basement", 'Region', player) and state.has("Power Star", player, world.MIPS1Cost[player].value))
     add_rule(world.get_location("MIPS 2", player), lambda state: state.can_reach("Basement", 'Region', player) and state.has("Power Star", player, world.MIPS2Cost[player].value))
 
-    world.completion_condition[player] = lambda state: state.can_reach("Bowser in the Sky", 'Region', player)
+    if world.CompletionType[player].value == 0:
```
Comment (Berserker66) [inline worlds/sm64ex/Rules.py:125]:
```suggestion
    if world.CompletionType[player]== "Last_Bowser_Stage":
```

File: worlds/sm64ex/Rules.py [hunk 2]
```diff
@@ -122,4 +122,9 @@ def set_rules(world, player: int, area_connections):
...
+        world.completion_condition[player] = lambda state: state.can_reach("Bowser in the Sky", 'Region', player)
+    elif world.CompletionType[player].value == 1:
```
Comment (Berserker66) [inline worlds/sm64ex/Rules.py:127]:
```suggestion
    elif world.CompletionType[player] == "All_Bowser_Stages":
```

**Comment by N00byKing** [comment]
Didn't know you can just access the option without `.value`, neat.

File: worlds/sm64ex/Rules.py [hunk 3]
```diff
@@ -122,4 +122,9 @@ def set_rules(world, player: int, area_connections):
     add_rule(world.get_location("MIPS 1", player), lambda state: state.can_reach("Basement", 'Region', player) and state.has("Power Star", player, world.MIPS1Cost[player].value))
     add_rule(world.get_location("MIPS 2", player), lambda state: state.can_reach("Basement", 'Region', player) and state.has("Power Star", player, world.MIPS2Cost[player].value))
 
-    world.completion_condition[player] = lambda state: state.can_reach("Bowser in the Sky", 'Region', player)
+    if world.CompletionType[player] == "Last_Bowser_Stage":
+        world.completion_condition[player] = lambda state: state.can_reach("Bowser in the Sky", 'Region', player)
+    elif world.CompletionType[player] == "All_Bowser_Stages":
```
Comment (Berserker66) [inline worlds/sm64ex/Rules.py:127]:
```suggestion
    if world.CompletionType[player] == "last_bowser_stage":
        world.completion_condition[player] = lambda state: state.can_reach("Bowser in the Sky", 'Region', player)
    elif world.CompletionType[player] == "all_bowser_stages":
```

Turns out I gave wrong advice, this should actually work.

**Comment by N00byKing** [comment]
Bump for re-review

**Review by NewSoupVi — APPROVED** [review]

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**